consensus/safety-rules/benches/safety_rules.rs

```diff
@@ -55,52 +69,56 @@ fn lsr(mut safety_rules: Box<dyn TSafetyRules<Vec<u8>>>, signer: ValidatorSigner
 fn in_memory(n: u64) {
     let signer = ValidatorSigner::from_int(0);
     let waypoint = test_utils::validator_signers_to_waypoints(&[&signer]);
+    let (prikey, pubkey) = test_utils::test_execution_keypair();
     let storage = PersistentSafetyStorage::initialize(
         Box::new(InMemoryStorage::new()),
         signer.private_key().clone(),
         waypoint,
     );
-    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage);
-    lsr(safety_rules_manager.client(), signer, n);
+    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage, pubkey);
+    lsr(safety_rules_manager.client(), signer, n, prikey);
 }
 
 fn on_disk(n: u64) {
     let signer = ValidatorSigner::from_int(0);
     let file_path = NamedTempFile::new().unwrap().into_temp_path().to_path_buf();
     let waypoint = test_utils::validator_signers_to_waypoints(&[&signer]);
+    let (prikey, pubkey) = test_utils::test_execution_keypair();
     let storage = PersistentSafetyStorage::initialize(
         Box::new(OnDiskStorage::new(file_path)),
         signer.private_key().clone(),
         waypoint,
     );
-    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage);
-    lsr(safety_rules_manager.client(), signer, n);
+    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage, pubkey);
+    lsr(safety_rules_manager.client(), signer, n, prikey);
 }
 
 fn serializer(n: u64) {
     let signer = ValidatorSigner::from_int(0);
     let file_path = NamedTempFile::new().unwrap().into_temp_path().to_path_buf();
     let waypoint = test_utils::validator_signers_to_waypoints(&[&signer]);
+    let (prikey, pubkey) = test_utils::test_execution_keypair();
     let storage = PersistentSafetyStorage::initialize(
         Box::new(OnDiskStorage::new(file_path)),
         signer.private_key().clone(),
         waypoint,
     );
-    let safety_rules_manager = SafetyRulesManager::new_serializer(signer.author(), storage);
-    lsr(safety_rules_manager.client(), signer, n);
+    let safety_rules_manager = SafetyRulesManager::new_serializer(signer.author(), storage, pubkey);
+    lsr(safety_rules_manager.client(), signer, n, prikey);
 }
 
 fn thread(n: u64) {
     let signer = ValidatorSigner::from_int(0);
     let file_path = NamedTempFile::new().unwrap().into_temp_path().to_path_buf();
     let waypoint = test_utils::validator_signers_to_waypoints(&[&signer]);
+    let (prikey, pubkey) = test_utils::test_execution_keypair();
     let storage = PersistentSafetyStorage::initialize(
         Box::new(OnDiskStorage::new(file_path)),
```

Comment:
it seems like this execution keypair should be pushed into initialize and not into the other apis

Reply:
PersistentSafetyStorage is a glue wrapper for SafetyRules around SecureStorage. The executor key will be stored in our storage solutions that allow a peer to safely access the public key without having access to the private key. We leverage it for all these type of activities and will see more usage as we expand our authenticated interfaces.